tst/EnergyPlus/unit/IndoorGreen.unit.cc

```diff
@@ -0,0 +1,225 @@
+        "    0.6;                        !- Radiant Fraction of LED Lights",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    Material::GetMaterialData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    HeatBalanceManager::GetConstructData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    HeatBalanceManager::GetZoneData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    state->dataSurfaceGeometry->CosZoneRelNorth.allocate(1);
+    state->dataSurfaceGeometry->SinZoneRelNorth.allocate(1);
+    state->dataSurfaceGeometry->CosZoneRelNorth(1) = std::cos(-state->dataHeatBal->Zone(1).RelNorth * Constant::DegToRadians);
+    state->dataSurfaceGeometry->SinZoneRelNorth(1) = std::sin(-state->dataHeatBal->Zone(1).RelNorth * Constant::DegToRadians);
+    state->dataSurfaceGeometry->CosBldgRelNorth = 1.0;
+    state->dataSurfaceGeometry->SinBldgRelNorth = 0.0;
+
+    SurfaceGeometry::GetSurfaceData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    state->dataHVACGlobal->TimeStepSys = 1;
+    state->dataHVACGlobal->TimeStepSysSec = state->dataHVACGlobal->TimeStepSys * Constant::SecInHour;
+
+    state->dataGlobal->NumOfTimeStepInHour = 1;
+    state->dataGlobal->MinutesPerTimeStep = 60 / state->dataGlobal->NumOfTimeStepInHour;
+    state->dataGlobal->TimeStep = 1;
+    state->dataGlobal->HourOfDay = 1;
+    state->dataEnvrn->DayOfWeek = 1;
+    state->dataEnvrn->DayOfYear_Schedule = 1;
+    ScheduleManager::UpdateScheduleValues(*state);
+
+    IndoorGreen::GetIndoorGreenInput(*state, ErrorsFound); // setup indoor living wall data
+    EXPECT_FALSE(ErrorsFound);                             // expect no errors
+    auto &thisindoorgreen = state->dataIndoorGreen->indoorGreens(IndoorGreenNum);
+    EXPECT_EQ(thisindoorgreen.LeafArea, 30.0);
+    EXPECT_EQ((int)thisindoorgreen.lightingMethod, (int)IndoorGreen::LightingMethod::LED);
+    EXPECT_EQ((int)thisindoorgreen.etCalculationMethod, (int)IndoorGreen::ETCalculationMethod::PenmanMonteith);
+}
+TEST_F(EnergyPlusFixture, IndoorGreen_CheckETFunction)
+{
+    bool ErrorsFound(false);
+    int IndoorGreenNum(1);
+    Real64 ZonePreTemp = 0.0; // C
+    Real64 ZonePreHum = 0.001;
+    Real64 LAI = 1.0;
+    state->dataIndoorGreen->indoorGreens.allocate(IndoorGreenNum);
+    auto &thisindoorgreen = state->dataIndoorGreen->indoorGreens(IndoorGreenNum);
+    thisindoorgreen.ZCO2 = 400;  // ppm
+    thisindoorgreen.ZPPFD = 0;   // umol/(m2s)
+    thisindoorgreen.ZVPD = 2000; // Pa
+    thisindoorgreen.ETRate =
+        IndoorGreen::ETPenmanMonteith(*state, ZonePreTemp, ZonePreHum, thisindoorgreen.ZCO2, thisindoorgreen.ZPPFD, thisindoorgreen.ZVPD, LAI);
+    EXPECT_EQ(thisindoorgreen.ETRate, 0.0);
+
+    thisindoorgreen.ETRate =
+        IndoorGreen::ETStanghellini(*state, ZonePreTemp, ZonePreHum, thisindoorgreen.ZCO2, thisindoorgreen.ZPPFD, thisindoorgreen.ZVPD, LAI);
+    EXPECT_EQ(thisindoorgreen.ETRate, 0.0);
+}
```

Comment:
This is a fine set of unit tests, although coverage could be better.  Your new test file IDF covers [54%](http://energyplus.s3-website-us-east-1.amazonaws.com/coverage/2024-02-12-EnergyPlus-0f65d6fc1e-PR10385-x86_64-gcc-11.4ICD/EnergyPlus/IndoorGreen.cc.gcov.html) of the code here, and your unit tests cover [28%](http://energyplus.s3-website-us-east-1.amazonaws.com/coverage/2024-02-12-EnergyPlus-0f65d6fc1e-PR10385-x86_64-gcc-11.4UTCD/EnergyPlus/IndoorGreen.cc.gcov.html) -- although many of the missed lines are error conditions.  You could consider whether to add some more cases to the unit tests here to cover more of the code, but it doesn't have to be a priority right now.

Reply:
Updated the unit test. Please let me know if you have addtional test in mind I can add in. Thank you!